src/main/java/com/grave/Game/Entities/Bullet.java

```diff
@@ -67,10 +67,14 @@ public void processAction(Action action) {
             flyTimer.reset();
 
             Vector3f playerLoc = objectManager.getEntity(fireAction.shooterID).getPosition();
-            Vector3f direction = fireAction.targetPosition.subtract(playerLoc).normalize();
+            Vector3f targetPos = fireAction.targetPosition;
+            targetPos.setZ(0);
+
```

Comment:
In dem Fall war der Fehler, dass z != 0 ist, oder was? Mir ist durchaus aufgefallen, dass die Werte merkwürdig hoch sind, aber so aus dem Stand hätte ich trotzdem gesagt, dass z doch null sein muss. Der Player bewegt sich doch ausschließlich auf der z=0 Ebene.

Reply:
Ja auch. Du nimmst ja hier die Mausposition als targetposition. Diese wird aber durch die MouseOnScreenPosition in die MouseInWorldPosition umgewandelt. Also bekommt die irgendeinen ‘random‘ Z-Wert, und somit wird der Vektor zwischen dem Player und der targetposition sehr groß. Und ich glaube irgendwo hast du mal vergessen was zu normalizen

Reply:
Ok, ich verstehe - macht Sinn